examples/creatives/hover_nonlinear.js

```diff
@@ -0,0 +1,141 @@
+        this.onHover_('content_container', 'mouseover');
+        this.onMouseOut_('content_container', 'mouseout');
+    }
+
+    /**
+     * Sends a SIMID message whenever an element is clicked.
+     * @param {String} elementName The name of the element.
+     * @param {String} message The message to send to the player.
+     * @private
+     */
+    sendMessageOnClick_(elementName, message) {
+        const sendMessageFunction = () => {this.simidProtocol.sendMessage(message);}
+        document.getElementById(elementName).addEventListener('click', sendMessageFunction);
+    }
+
+    /**
+     * Adds a hover event listener to the contents of the iframe that expands the iframe.
+     * @param {String} elementName The name of the element.
+     * @param {Event} event The event performed on the element.
+     * @private
+     */   
+    onHover_(elementName, event) {
+        const expandOnHoverFunction = () => {
+            const newDimensions = {};
+            let resizeParams = {};
+
+            let desiredX;
+            let desiredWidth;
+            let desiredY;
+            let desiredHeight;
+            let fullCreativeWidth;
+            let fullCreativeHeight;
+
+            //Width grows 20% 
+            desiredWidth = this.initialDimensions_.width * DIMENSION_GROWTH;
+            desiredX = parseInt(this.initialDimensions_.x) - parseInt(this.initialDimensions_.width * OFFSET_SHIFT);
+
+            //Makes sure x offset fits in the player
+            if (desiredX <= 0) {
+                desiredX = 0;
+                desiredWidth = this.initialDimensions_.width * DGE_DIMENSION_GROWTH;
+            }
+            if (desiredX > this.videoDimensionWidth_) {
+                return;
+            }
+
+            //Makes sure width fits in the player
+            fullCreativeWidth = desiredX + desiredWidth;
+            if (fullCreativeWidth > this.videoDimensionWidth_) {
+                fullCreativeWidth = this.videoDimensionWidth_;
+                desiredWidth = fullCreativeWidth - desiredX;
+            }
+            
+            //Height grows 20%
+            desiredHeight = this.initialDimensions_.height * DIMENSION_GROWTH;
+            desiredY = parseInt(this.initialDimensions_.y) - parseInt(this.initialDimensions_.height * OFFSET_SHIFT);
+
+            //Makes sure y offset fits in the player
+            if (desiredY <= 0) {
+                desiredY = 0;
+                desiredHeight = this.initialDimensions_.height * DGE_DIMENSION_GROWTH;
+            }
+            if (desiredY > this.videoDimensionHeight_) {
+                return;
+            }
+
+            //Makes sure height fits in the player
+            fullCreativeHeight = desiredY + desiredHeight;
+            if (fullCreativeHeight > this.videoDimensionHeight_) {
+                fullCreativeHeight = this.videoDimensionHeight_;
+                desiredHeight = fullCreativeHeight - desiredY;
+            }
+
```

Comment:
Yeah, this looks a lot like what we discussed. It should "just work" in all situations. You might not get as good of an effect with the postage stamp non linear, but that could be an edge case.

Reply:
the "postage stamp" expands to just a bigger postage stamp, which I think is good for the purpose of our sample creative